app/controllers/categories_controller.rb

```diff
@@ -1,2 +1,48 @@
 class CategoriesController < ApplicationController
+  def new
+    render locals: {
+      category: Category.new
+    }
+  end
+
+  def create
+    category = Category.new
+    category.name = params[:category][:name]
+
+    if category.save
+      redirect_to actions: "index"
+    else
+      flash[:alert] = "Category could not be saved."
```

Comment:
Our users will get better messaging if this line is
`flash[:alert] = category.errors